src/core/misc.c

```diff
@@ -719,6 +719,13 @@ int expand_escape(const char **data)
 			}
 		}
 		return strtol(digit, NULL, 8);
+
+	case '\\':
+		/* escaped sequence, if the next character isn't part of a valid
```

Comment:
I mean instead of all that code, technically we should be expanding "\\\\" into "\\"